op_builder/async_io.py

```diff
@@ -59,6 +90,11 @@ def is_compatible(self):
         aio_compatible = self.has_function('io_submit', ('aio', ))
         if not aio_compatible:
             self.warning(
-                f"{self.NAME} requires libaio but it is missing. Can be fixed by: `apt install libaio-dev`."
+                f"{self.NAME} requires libaio but it was not found.  If libaio is installed, try setting CFLAGS and LDFLAGS."
             )
+
+            # Check for the libaio package via known package managers
+            # to print suggestions on which package to install.
+            self.check_for_libaio_pkg()
```

Comment:
```suggestion
                f"{self.NAME} requires the dev libaio .so object and headers but these were not found. 
            )

            # Check for the libaio package via known package managers
            # to print suggestions on which package to install.
            self.check_for_libaio_pkg()
            
            self.warning("If libaio is already installed, (perhaps from source), try setting CFLAGS and LDFLAGS environment variables to where it can be found.")
```

I'd suggest to first give users a possible easy solution via their package manager and only then the fallback to the complicated solution.

But I also don't fully understand the logic here - if the right package is found already to be installed why would it fail to build `io_submit`. User messing up their global `ld.conf`?

If we checked that it's already found to be installed, should we not say that? rather than "if libaio is already installed"

Reply:
Thanks for reviewing this, @stas00 .  If libaio was installed from source or through something like conda, it might actually be installed, but just not in a path where the compiler looks by default.  The CFLAGS/LDFLAGS tip is trying to help users who are in that situation.

The combination of the failed package manager check and this warning just after would produce a pair of messages like:
```
async_io: please install the libaio-devel package with rpm
If libaio is already installed, (perhaps from source), try setting CFLAGS and LDFLAGS environment variables to where it can be found.
```

Reply:
I totally get the ` CFLAGS/LDFLAGS ` case - I probably wasn't clear if it sounded to you that I thought it was redundant. That's why I thought you're in the best position to complete this, since yours is that special case, Adam.

Reply:
but actually it'll produce 3 lines:
```
async_io requires libaio but it was not found
please install the libaio-devel package with rpm
If libaio is already installed, (perhaps from source), try setting CFLAGS and LDFLAGS environment variables to where it can be found.

```

Reply:
Yes, I just realized that I missed that third line you suggested.  Those changes should be in there now.

Reply:
I still need to add a check for the C compiler.